core-js/src/main/javascript/cdf/components/ccc/BaseCccComponent.js

```diff
@@ -77,21 +97,109 @@ define([
       }
 
       if(typeof Modernizr !== 'undefined' && Modernizr.svg) {
-        this.renderChart();
+        this.render();
       } else {
         pv.listenForPageLoad(function() {
-          me.renderChart();
+          me.render();
         });
       }
     },
 
+    _getCCCName: function(name) {
+      var regex = /(.*)(Chart)$/g;
+      var matching = name.split(regex);
+      if (matching.length > 1) {
+        return matching[1].substr( 0, 1 ).toLowerCase() + matching[1].substr( 1 );
+      }
+    },
+
+    _getCCCFullName: function(name) {
+      var fullName = this._getCCCName(name);
+      if (this.chartDefinition) {
+        var cd = this.chartDefinition;
+        // exceptions
+        if (fullName.indexOf("area") > -1 || fullName == "stackedArea") {
+          return "areaStacked";
+        } else if (fullName == "metricDot") {
+          return "bubble";
+        } else if (fullName == "metricLine") {
+          return "scatter";
+        } else if (fullName == "normalizedBar") {
+          return "barNormalized";
+        } else if (fullName == "dot" || fullName == "stackedDot") {
+          return "pointAbstract";
+        } else if (fullName == "stackedLine") {
+          return "line";
+        }
+
+        // general cases
+        if (cd.valuesNormalized) {
+          fullName += "Normalized";
+        }
+        if (cd.orientation === "Horizontal") {
+          fullName += cd.orientation;
+        }
+        if (cd.stacked == true || fullName == "area") {
+          fullName += "Stacked";
+        }
+      }
+
+      return fullName;
+    },
+
     render: function(values) {
+      var myself = this;
```

Comment:
On another end, I would not remove the base class' `renderChart` method: there are dashboards out there overriding this (not very orthodox, I know) and I think there might be other ways to handle this.

Here's an alternative that I think would work better:

```js
var chart = {
  renderChart: function() {		
    
    // Always executes asynchronously.

    var cd = this.chartDefinition;		
    
    if(this.dashboard.isValidQueryDefinition(cd)) {
      
      this.beginQuery(cd, _.bind(this._renderOuter, this));

    } else if(this.valuesArray != null) {

      this.execute(_.bind(function() { this._renderOuter(this.valuesArray); }, this));

    } else {
      // Initialize the component only
      this.execute(_.bind(this._renderOuter, this));
    }
  },

  _shouldUseVizApiStyles: function() {
    return !!this._vizApiStyles && !_.contains(chartTypesBlackList, this.getCccChartName());
  },

  _renderOuter: function(data) {

    var promiseExtensions;
       
    this._preProcessChartDefinition();

    if(this._shouldUseVizApiStyles()) {

      promiseExtensions = context.getAsync("pentaho/ccc/visual/" + this.getCccChartName())
          .then(function(View) {
             return View.type.extensionEffective;
          });
    } else {
      // no external extensions
      promiseExtensions = Promise.resolve(null);
    }

    promiseExtensions
      .then(_.bind(this._renderInner, this, data))
      .then(_.bind(this.endExec, this), _.bind(this.failExec, this));
  },

  render: function(data) {
    this._renderInner(data, null);
  },

  // Similar to yours `_internalRender` and to the old "render" method.
  _renderInner: function(data, externalChartDefinition) {
    
    $("#" + myself.htmlObject).append('<div id="' + myself.htmlObject + 'protovis"></div>');

    // Always clone the original chartDefinition.
    var cd = $.extend({}, this.chartDefinition);

    if(externalChartDefinition) cd = $.extend(externalChartDefinition, cd);

    cd.canvas = this.htmlObject + "protovis";

    // Process extension points
    if(cd.extensionPoints) {
      var ep = {};
      cd.extensionPoints.forEach(function(a) { ep[a[0]] = a[1]; });
      cd.extensionPoints = ep;
    }

    if(this._shouldUseVizApiStyles() && cd.colors == null) {
      cd.colors = paletteRegistry.get("cdf_default").colors;
    }

    this.chart = new this.cccType(cd);

    if(arguments.length > 0) {
      this.chart.setData(data, {crosstabMode: this.crosstabMode, seriesInRows: this.seriesInRows});
    }

    this.chart.render();
  }
}
```